go/detectparser/fortigate_dhcp.go

```
@@ -0,0 +1,83 @@
+package detectparser
+
+import (
+	"regexp"
+
+	"github.com/inverse-inc/go-utils/sharedutils"
+)
+
+var fortiGateDhcpRegexPattern1 = regexp.MustCompile(`(\w+)="([^"]*)"|(\w+)=([^\s]+)`)
+
+type FortiGateDhcpParser struct {
+	Pattern1, Pattern2 *regexp.Regexp
+	parser
+}
+
+func (s *FortiGateDhcpParser) Parse(line string) ([]ApiCall, error) {
+	matches := s.Pattern1.FindAllStringSubmatch(line, -1)
+	var mac, ip, lease, hostname, ack string
+	var err error
+
+	attributes := make(map[string]string)
+
+	for _, match := range matches {
+		if match[1] != "" {
+			attributes[match[1]] = match[2]
+		} else {
+			attributes[match[3]] = match[4]
+		}
+	}
+
+	for key, value := range attributes {
+		if key == "mac" {
+			mac = value
+		} else if key == "ip" {
+			ip = value
+		} else if key == "lease" {
+			lease = value
+		} else if key == "hostname" {
+			hostname = value
+		} else if key == "dhcp_msg" {
+			ack = value
+		}
+	}
+
+	if ack == "" || ack != "Ack" {
+		return nil, nil
+	}
+
+	if ip, err = sharedutils.CleanIP(ip); err != nil {
+		return nil, nil
```

Should we return an error?
```go
return nil, fmt.Errorf("Invalid ip %s: %w", ip, err)
```